src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/split_fc.cpp

```diff
@@ -0,0 +1,213 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/core/rt_info.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "openvino/pass/constant_folding.hpp"
+#include <transformations/utils/utils.hpp>
+#include "openvino/op/concat.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/convert.hpp"
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/reshape.hpp"
+#include "openvino/op/subtract.hpp"
+#include "openvino/op/variadic_split.hpp"
+#include "transformations/cpu_opset/common/op/fully_connected.hpp"
+
+#include "split_fc.hpp"
+
+#include "itt.hpp"
+
+ov::intel_cpu::SplitFC::SplitFC(int sub_stream_num) {
+    MATCHER_SCOPE(SplitFC);
+    auto fc_m = ov::pass::pattern::wrap_type<ov::intel_cpu::FullyConnectedNode>();
+
+    ov::matcher_pass_callback callback = [=](ov::pass::pattern::Matcher& m) {
+        const auto& pattern_map = m.get_pattern_value_map();
+
+        const auto& fc_node = pattern_map.at(fc_m).get_node_shared_ptr();
+        auto& rt_info = fc_node->get_rt_info();
+        if (rt_info.count("parallelDomain")) {
+            return false;
+        }
+
+        auto src_item = fc_node->get_input_node_shared_ptr(0);
+        auto fc_weight_node = fc_node->get_input_node_shared_ptr(1);
+
+        // split happens on the first dimension.
+        constexpr size_t split_dim = 0;
+        auto split_dim_node = std::make_shared<ov::op::v0::Constant>(ov::element::i32, ov::Shape{}, split_dim);
+
+        // parts will be splited according the sub stream num.
+        int split_num = sub_stream_num + 1;
+
+        auto split_parts = [](int len, int n) {
+            int average = len / n;
+            std::vector<int> parts(n, average);
+            parts.back() = len - average * (n - 1);
+            return parts;
+        };
+
+        // TODO: support transpose
+        if (ov::is_type<ov::op::v1::Transpose>(fc_weight_node)) {
```

Comment:
Current implementation has the following limitations:

1. Transpose on weights is not supported.
2. Decompression without Subtract is not supported.
3. Decompression with Subtract without convert (with f32 constant) is not supported.
4. This transformation breaks postops/bias fusions for FullyConnected node.

All these limitations "pop up" when [MatmulWeightsDecompression](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/matmul_weights_decompression.cpp) subgraph tests are running on multi socket platform: the corresponding tests fail.
For each limitation we need to decide whether it must be eliminated within this PR or the corresponding improvements can be implemented in follow-up PRs. If some of the limitations will not be eliminated in this PR, we need to decide what to do with the failed tests: we should probably skip them

CC @dmitry-gorokhov